### PR TITLE
New module Sourcepredict

### PR DESCRIPTION
**Pull request by MeriamOs**
## PR checklist

Add sourcepredict module

- [x] If you've added a new tool - have you followed the module conventions in the [contribution docs](https://github.com/nf-core/modules/tree/master/.github/CONTRIBUTING.md)
- [x] If necessary, include test data in your PR.
- [x] Remove all TODO statements.
- [x] Emit the `versions.yml` file.
- [x] Follow the naming conventions.
- [x] Follow the parameters requirements.
- [x] Follow the input/output options guidelines.
- [x] Add a resource `label`
- [x] Use BioConda and BioContainers if possible to fulfil software requirements.
- Ensure that the test works with either Docker / Singularity. Conda CI tests can be quite flaky:
  - For modules:
    - [x] `nf-core modules test <MODULE> --profile docker`
    - [x] `nf-core modules test <MODULE> --profile singularity`
    - [x] `nf-core modules test <MODULE> --profile conda`

### CODE REVIEW

File: modules/nf-core/sourcepredict/main.nf [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+    input:
+    tuple val(meta), path(kraken_parse)
+    path sources
+    path labels
+    path taxa_sqlite, stageAs:'.etetoolkit/*'
+    path taxa_sqlite_traverse_pkl, stageAs:'.etetoolkit/*'
+
+    output:
+    tuple val(meta), path("*.sourcepredict.csv"), emit: report
+    path "versions.yml"           , emit: versions
```
Comment (fellen31) [inline modules/nf-core/sourcepredict/main.nf]: align

File: modules/nf-core/sourcepredict/main.nf [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+    conda "${moduleDir}/environment.yml"
+    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
+        'https://depot.galaxyproject.org/singularity/sourcepredict:0.5.1--pyhdfd78af_0':
+        'biocontainers/sourcepredict:0.5.1--pyhdfd78af_0' }"
+
+    input:
+    tuple val(meta), path(kraken_parse)
+    path sources
+    path labels
+    path taxa_sqlite, stageAs:'.etetoolkit/*'
```
Comment (sateeshperi) [inline modules/nf-core/sourcepredict/main.nf:14]:
```suggestion
    path(taxa_sqlite, stageAs: '.etetoolkit/*')
```

File: modules/nf-core/sourcepredict/main.nf [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+    path taxa_sqlite_traverse_pkl, stageAs:'.etetoolkit/*'
```
Comment (sateeshperi) [inline modules/nf-core/sourcepredict/main.nf:15]:
```suggestion
    path(taxa_sqlite_traverse_pkl, stageAs: '.etetoolkit/*')
```

File: modules/nf-core/sourcepredict/main.nf (hunk 1 above)
Comment (sateeshperi) [inline modules/nf-core/sourcepredict/main.nf:19]:
```suggestion
    tuple val(meta), path("*.sourcepredict.csv"), emit: report
    path "versions.yml"                                       , emit: versions
```

File: modules/nf-core/sourcepredict/main.nf [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+    tuple val(meta), path("*.sourcepredict.csv"), emit: report
+    path "versions.yml"           , emit: versions
+
+    when:
+    task.ext.when == null || task.ext.when
+
+    script:
+    def args = task.ext.args ?: ''
+    def prefix = task.ext.prefix ?: "${meta.id}"
+
```
Comment (sateeshperi) [inline modules/nf-core/sourcepredict/main.nf:27]:
```suggestion
```

File: modules/nf-core/sourcepredict/main.nf [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+    cat <<-END_VERSIONS > versions.yml
+    "${task.process}":
+        sourcepredict: \$(python -c "import sourcepredict; print(sourcepredict.__version__)")
+    END_VERSIONS
+    """
+
+    stub:
+    def args = task.ext.args ?: ''
+    def prefix = task.ext.prefix ?: "${meta.id}"
+
```
Comment (sateeshperi) [inline modules/nf-core/sourcepredict/main.nf:49]:
```suggestion
```

File: modules/nf-core/sourcepredict/main.nf [hunk 6]
```diff
@@ -0,0 +1,58 @@
...
+    when:
+    task.ext.when == null || task.ext.when
+
+    script:
+    def args = task.ext.args ?: ''
+    def prefix = task.ext.prefix ?: "${meta.id}"
+
+    """
+    export NUMBA_CACHE_DIR='/tmp'
```
Comment (fellen31) [inline modules/nf-core/sourcepredict/main.nf:28]: Writing to `/tmp` can break some clusters, could you write this to `./tmp` or `.` i instead, so a temporary folder is created within the work directory?

File: modules/nf-core/sourcepredict/tests/main.nf.test [hunk 7]
```diff
@@ -0,0 +1,88 @@
...
+    process "SOURCEPREDICT"
+    config "./nextflow.config"
+    tag "modules"
+    tag "modules_nfcore"
+    tag "xz/decompress"
+    tag "sourcepredict"
+
+    setup {
+        run("XZ_DECOMPRESS") {
+            script "modules/nf-core/xz/decompress/main.nf"
```
Comment (sateeshperi) [inline modules/nf-core/sourcepredict/tests/main.nf.test:14]: replace with relative path here plz

File: modules/nf-core/sourcepredict/meta.yml [hunk 8]
```diff
@@ -0,0 +1,70 @@
...
+
+  - - labels:
+        type: file
+        description: Labels for the sources table in csv format. Default can be downloaded from
+          https://raw.githubusercontent.com/maxibor/sourcepredict/master/data/modern_gut_microbiomes_labels.csv
+        pattern: "*.csv"
+
+  - - taxa_sqlite:
+        type: file
+        description: taxa.splite file downloaded with ete3 toolkit
```
Comment (fellen31) [inline modules/nf-core/sourcepredict/meta.yml:42]: No pattern?

File: modules/nf-core/sourcepredict/tests/main.nf.test [hunk 9]
```diff
@@ -0,0 +1,88 @@
...
+        }
+
+        then {
+            assertAll(
+                { assert process.success },
+                { assert snapshot(
+                    process.out.report,
+                    process.out.versions,
+                ).match()
+                }
```
Comment (fellen31) [inline modules/nf-core/sourcepredict/tests/main.nf.test:50]: If everything can be snapshot:

```suggestion
                { assert snapshot(process.out).match() }
```

File: modules/nf-core/sourcepredict/tests/main.nf.test [hunk 10]
```diff
@@ -0,0 +1,88 @@
...
+                input[3] = XZ_DECOMPRESS.out.file.map{ it[1] }
+                input[4] = file(params.modules_testdata_base_path + 'genomics/prokaryotes/metagenome/taxonomy/misc/taxa_sqlite_traverse.pkl', checkIfExists: true)
+                """
+            }
+        }
+
+        then {
+            assertAll(
+                { assert process.success },
+                { assert snapshot(
```
Comment (fellen31) [inline modules/nf-core/sourcepredict/tests/main.nf.test:78]: Same here

File: modules/nf-core/sourcepredict/main.nf (hunk 6 above)
Comment (sateeshperi) [reply modules/nf-core/sourcepredict/main.nf:28]: or if the tmp can be provided via `ext.args` that would also do

Comment (MeriamOs) [reply modules/nf-core/sourcepredict/main.nf]: Changed to ./tmp

File: modules/nf-core/sourcepredict/tests/main.nf.test (hunk 10 above)
Comment (MeriamOs) [reply modules/nf-core/sourcepredict/tests/main.nf.test]: Fixed

**Review by sateeshperi — APPROVED** [review]

**Pull request merged**